docs/latest/design/ratelimit.md

```diff
@@ -0,0 +1,258 @@
+            kind: RateLimitFilter
+            name: ratelimit-per-user
+	- type: ExtensionRef
+          extensionRef:
+            group: gateway.envoyproxy.io
+            kind: RateLimitFilter
+            name: ratelimit-all-safeguard-app    
+      backendRefs:
+        - name: backend
+          port: 3000
+```
+
+* The user has created two `RateLimitFilter`s  and has attached it to a `HTTPRoute` - one(`ratelimit-all-safeguard-app`) to
+ensure that the backend does not get overwhelmed with requests, any excess requests are rate limited irrespective of
+the attributes within the traffic flow, and another(`ratelimit-per-user`) to rate limit each distinct user client
+who can be differentiated using the `x-user-id` header, to ensure that each client does not make exessive requests to the backend.
+* If user `baz` (identified with the header and value of `x-user-id: baz`) sends 90 requests within the first second, and
+user `bar` sends 11 more requests during that same interval of 1 second, and user `bar` sends the 101th request within that second,
+the rule defined in `ratelimit-all-safeguard-app` gets activated and Envoy Gateway will ratelimit the request sent by `bar` (and any other
+request sent within that 1 second). After 1 second, the rate limit counter associated with the `ratelimit-all-safeguard-app` rule
+is reset and again evaluated.
+* If user `bar` also ends up sending 90 more requests within the hour, summing up `bar`'s total request count to 101, the rate limit rule
+defined within `ratelimit-per-user` will get activated, and `bar`'s requests will be rate limited again until the hour interval ends.
+* Within the same above hour, if `baz` sends 11 more requests, summing up `baz`'s total request count to 101, the rate limit rule defined
+within `ratelimit-per-user` will get activated for `baz`, and `baz`'s requests will also be rate limited until the hour interval ends. 
+
+## Design Decisions
+
+* The initial design uses an Extension filter to apply the Rate Limit functionality on a specific `HTTPRoute`.
+This was preferred over the PolicyAttachment Extension mechanism, because it is unclear whether Rate Limit
+will be required to be enforced or overridden by the platform administrator or not.
+* The RateFilter can only be applied as a filter to a `HTTPRouteRule`, applying it across all backends within a `HTTPRoute`
+and cannot be applied a filter within a `HTTPBackendRef` for a specific backend.
+* The HTTPRoute API has a `matches` field within each `rule` to select a specific traffic flow to be routed to
+the destination backend. The RateLimitFilter API that can be attached to an HTTPRoute via an `extensionRef` filter,
+also has a `matches` field within each `rule` to select attributes within the traffic flow to rate limit specific clients.
+The two levels of `matches` allow for flexibility and aim to hold match information specific to its use, allowing the author/owner
+of each configuration to be different.
```

Comment:
Regarding:
>allowing the author/owner of each configuration to be different.

I would like users to tell us this is a requirement b/c having `matches` in a route and the RL filter. I suggest we start simple and have the RL filter express rate limits.

Reply:
as stated here and above, `matches` in the RL filter is an important field in the initial design that is not just good to have, but is needed